extensions/ql-vscode/src/common/vscode/telemetry.ts

```diff
@@ -297,3 +297,21 @@ export async function initializeTelemetry(
   ctx.subscriptions.push(telemetryListener);
   return telemetryListener;
 }
+
+function isGlobalTelemetryEnabled(): boolean {
```

Comment:
The docs suggest not trying to implement this logic ourselves and instead using `vscode.env.isTelemetryEnabled`. I haven't yet looked into whether we can actually do that and it does what we want, or if we do indeed have to check both config settings.

Reply:
This method is complicated enough that it would be nice to use existing logic for it.

Reply:
I've done some testing of how `isTelemetryEnabled` behaves. The answer is:
- It does indeed take into account both `telemetry.enableTelemetry` and `telemetry.telemetryLevel`
- It only returns true if telemetry is FULLY enabled. So that means `telemetryLevel` is set to `all`. If the level is set to `error` then `isTelemetryEnabled` will return false.

Maybe this is fine for us and the CodeQL extension doesn't need the granularity of only sending error reports and not other telemetry. We certainly aren't losing anything with this PR because we weren't checking the level at all before.

Are people happy with this limitation that if `telemetryLevel` is set to `error` then we won't send any telemetry?